### PR TITLE
chore: remove che-theia editor and its references from plugin reg

### PR DESCRIPTION
**Pull request by svor**
### What does this PR do?
Remove che-theia editor and its references from the plugin registry

The result could be checked here https://pr-check-930-devspaces-plugin-registry.surge.sh/

### What issues does this PR fix or reference?
https://issues.redhat.com/browse/CRW-4135

### CODE REVIEW

File: dependencies/che-plugin-registry/README.md [hunk 1]
```diff
@@ -1,6 +1,6 @@
 # Red Hat OpenShift Dev Spaces (formerly CodeReady Workspaces) Plugin Registry
 
-This repository holds ready-to-use plugins for different languages and technologies. 
+This repository holds editor definitions and ready-to-use plugins for different languages and technologies as part of the embedded instance of the [Open VSX](https://open-vsx.org/about) registry to support air-gapped, offline, and proxy-restricted environments. The embedded Open VSX registry contains only a subset of the extensions published on open-vsx.org that can be used with Microsoft Visual Studio Code editor.
```
Comment (nickboldt) [inline dependencies/che-plugin-registry/README.md:3]:
```suggestion
This repository holds editor definitions and ready-to-use plugins for different languages and technologies as part of the embedded instance of the [Open VSX](https://open-vsx.org/about) registry to support air-gapped, offline, and proxy-restricted environments. The embedded Open VSX registry contains only a subset of the extensions published on open-vsx.org that can be used with Microsoft Visual Studio Code - Open Source IDE.
```

**Comment by nickboldt** [comment]
What about these two GH actions? If you delete `che-theia-plugins.yaml` do we need to key on a different set of files? 

```
** [1] ./workflows/plugin-registry-build.yaml ** 
60-            /tmp/che-plugin-registry/download-folder
61-            /tmp/che-plugin-registry/unpack-folder
62:          key: download-unpacked-cache-${{ hashFiles('**/che-theia-plugins.yaml') }}
63-          restore-keys: |
64-            download-unpacked-cache-

** [2] ./workflows/plugin-registry-pr-check-build.yaml ** 
49-          /tmp/che-plugin-registry/download-folder
50-          /tmp/che-plugin-registry/unpack-folder
51:        key: download-unpacked-cache-${{ hashFiles('**/che-theia-plugins.yaml') }}
52-        restore-keys: |
53-          download-unpacked-cache-
```

**Comment by svor** [comment]
> What about these two GH actions? If you delete che-theia-plugins.yaml do we need to key on a different set of files?
I believe we can delete these blocks since we don't need to cache vsix dependencies anymore. Updated, thx!

**Review by nickboldt — APPROVED** [review]
if the only change since my last review was was removing that leftover stuff in the GH action yamls, then yes, +1.

**Pull request merged**